resilience4j-micronaut/src/main/java/io/github/resilience4j/mirconaut/retry/RetryInterceptor.java

```diff
@@ -0,0 +1,109 @@
+/*
+ * Copyright 2019 Michael Pollind
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.github.resilience4j.mirconaut.retry;
+
+import io.github.resilience4j.mirconaut.BaseInterceptor;
+import io.github.resilience4j.mirconaut.ResilienceInterceptPhase;
+import io.github.resilience4j.mirconaut.fallback.UnhandledFallbackException;
+import io.github.resilience4j.retry.Retry;
+import io.github.resilience4j.retry.RetryRegistry;
+import io.github.resilience4j.retry.transformer.RetryTransformer;
+import io.micronaut.aop.MethodInterceptor;
+import io.micronaut.aop.MethodInvocationContext;
+import io.micronaut.context.BeanContext;
+import io.micronaut.context.annotation.Requires;
+import io.micronaut.core.annotation.AnnotationValue;
+import io.micronaut.core.async.publisher.Publishers;
+import io.micronaut.core.convert.ConversionService;
+import io.micronaut.core.type.ReturnType;
+import io.micronaut.inject.ExecutableMethod;
+import io.micronaut.inject.MethodExecutionHandle;
+import io.reactivex.Flowable;
+
+import javax.inject.Singleton;
+import java.util.Optional;
+import java.util.concurrent.CompletionStage;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+
+@Singleton
+@Requires(beans = RetryRegistry.class)
+public class RetryInterceptor extends BaseInterceptor implements MethodInterceptor<Object,Object> {
+    private final RetryRegistry retryRegistry;
+    private final BeanContext beanContext;
+    private static final ScheduledExecutorService retryExecutorService = Executors.newScheduledThreadPool(Runtime.getRuntime().availableProcessors());
```

Comment:
This thread pool should be shutdown in a `@PreDestroy` method. Alternatively if you can use one of Micronaut's built in thread pools by injecting for example `@Named(TasksExecutors.IO) ScheduledExecutorService executorService`